src/spider/worker/TaskExecutor.hpp

```diff
@@ -33,19 +32,30 @@ enum class TaskExecutorState : std::uint8_t {
 
 class TaskExecutor {
 public:
-    TaskExecutor(
+    static auto spawn_cpp_executor(
```

Comment:
```suggestion
    [[nodiscard]] static auto spawn_cpp_executor(
```
Please add this to your PR checklist.